apps/dashboard/app/models/current_user.rb

```diff
@@ -20,9 +20,44 @@ class << self
 
   def initialize
     @pwuid = Etc.getpwuid
+    @user_settings = read_user_settings
   end
 
   def primary_group
     @primary_group ||= Etc.getgrgid(gid).name
   end
+
+  def user_settings
+    @user_settings.clone
+  end
+
+  def update_user_settings(new_user_settings)
+    @user_settings.deep_merge!(new_user_settings.deep_symbolize_keys)
+    save_user_settings
+  end
+
+  private
+  def read_user_settings
+    user_settings = {}
+    return user_settings unless user_settings_path.exist?
+
+    begin
+      yml = YAML.safe_load(user_settings_path.read) || {}
+      user_settings = yml.deep_symbolize_keys
+    rescue => e
+      Rails.logger.error("Can't read or parse settings file: #{user_settings_path} because of error #{e}")
+    end
+
+    user_settings
+  end
+
+  def save_user_settings
+    # Ensure there is a directory to write the user settings file
+    user_settings_path.dirname.tap { |p| p.mkpath unless p.exist? }
+    File.open(user_settings_path.to_s, "w") { |file| file.write(@user_settings.deep_stringify_keys.to_yaml) }
+  end
+
+  def user_settings_path
+    Pathname.new(::Configuration.dataroot).join(ENV['OOD_USER_SETTINGS'] || '.ood')
```

Comment:
can you put `OOD_USER_SETTINGS` into `string_configs` below as `user_settings`. It'll still respond to the ENV variable `OOD_USER_SETTINGS`.  We have a lot of miscellaneous ENV variables floating around in the code base, and so I'm trying to coalesce them all here.

https://github.com/OSC/ondemand/blob/fcd4a5d655d005641f4782ecadb8d3611812aade/apps/dashboard/config/configuration_singleton.rb#L53

Reply:
Updated to use the string_configs